pkg/flowaggregator/flowaggregator.go

```diff
@@ -724,3 +815,110 @@ func (fa *flowAggregator) createInfoElementForTemplateSet(ieName string, enterpr
 	}
 	return ie, nil
 }
+
+func (fa *flowAggregator) watchConfiguration() {
+	klog.Info("Watching for FlowAggregator configuration file")
+	for {
+		select {
+		case event, ok := <-fa.configWatcher.Events:
+			if !ok {
+				return
+			}
+			// When mounted configmap is updating, watcher event "CHMOD" and "REMOVE" will happen.
+			// Need to add configFile back to watcher in order to watch the updated configMap.
+			if event.Op == fsnotify.Remove {
+				if err := fa.configWatcher.Add(fa.configFile); err != nil {
+					klog.Fatal(err)
```

Comment:
I think we should avoid calling `Fatal` except in initialization code. I think if you look at the rest of the codebase, you'll see that we typically don't call `log.Fatal` (after initialization) and we don't panic.
IMO, a better option would be to return, and wrap the function with some exponential backoff logic. Something like this:
```go
func (fa *flowAggregator) watchConfiguration() {
    // start the watch
    // read the file for the first time
    for {
        select {
        case event, ok := <-fa.configWatcher.Events:
            if somethingBad {
                return err
            }
            if invalidConfig {
                klog.Error(...)
                break
            }
        }
    }
}
```
and the function can be used like this:
```go
wait.BackoffUntil(
    fa.watchConfiguration,
    wait.NewExponentialBackoffManager(1 * time.Second /* initBackoff */, 1 * time.Minute /* maxBackoff */, 5 * time.Minute /* resetbackoff */, 2 /* factor */, 0 /* jitter */, &clock.RealClock),
    stopCh,
)
```

Reply:
BTW, I think it could be a good idea to unit test this function if possible. I don't think you have added unit tests for this patch?
A good way to unit test this IMO could be to user https://github.com/spf13/afero#memmapfs, as Afero is already an Antrea dependency. You can make changes to the ConfigMap file (in memory file, appropriate for unit testing), and maybe check that the correct messages show up on the updateCh. Or you may have better ideas for unit tests.

Reply:
I've separated the part that does the update work and add unit test for updateFlowAggregator.